packages/mock-block-dock/src/mock-block-dock.tsx

```diff
@@ -170,70 +175,60 @@ export const MockBlockDock: FunctionComponent<MockBlockDockProps> = ({
 
   useEffect(() => {
     if (graphService) {
-      graphService.readonly({ data: debugReadonly });
+      graphService.readonly({ data: readonly });
     }
-  }, [debugReadonly, graphService]);
-
-  if (readonly !== prevReadonly.current && readonly !== debugReadonly) {
-    setDebugReadonly(!!readonly);
-  }
-
-  if (!debug) {
-    return (
-      <div ref={wrapperRef}>
-        {graphService ? (
-          <BlockRenderer
-            customElement={
-              "customElement" in blockDefinition
-                ? blockDefinition.customElement
-                : undefined
-            }
-            html={"html" in blockDefinition ? blockDefinition.html : undefined}
-            properties={propsToInject}
-            ReactComponent={
-              "ReactComponent" in blockDefinition
-                ? blockDefinition.ReactComponent
-                : undefined
-            }
-          />
-        ) : null}
-      </div>
-    );
-  }
+  }, [readonly, graphService]);
+
+  const Component = (
+    <div ref={wrapperRef}>
+      {graphService ? (
+        <BlockRenderer
+          customElement={
+            "customElement" in blockDefinition
+              ? blockDefinition.customElement
+              : undefined
+          }
+          html={"html" in blockDefinition ? blockDefinition.html : undefined}
+          properties={propsToInject}
+          ReactComponent={
+            "ReactComponent" in blockDefinition
+              ? blockDefinition.ReactComponent
+              : undefined
+          }
+        />
+      ) : null}
+    </div>
+  );
 
   return (
-    <Layout blockType={blockType}>
-      <Box>
-        <Box padding={3.75}>
-          <div ref={wrapperRef}>
-            {graphService ? (
-              <BlockRenderer
-                customElement={
-                  "customElement" in blockDefinition
-                    ? blockDefinition.customElement
-                    : undefined
-                }
-                html={
-                  "html" in blockDefinition ? blockDefinition.html : undefined
-                }
-                properties={propsToInject}
-                ReactComponent={
-                  "ReactComponent" in blockDefinition
-                    ? blockDefinition.ReactComponent
-                    : undefined
-                }
-              />
-            ) : null}
-          </div>
-        </Box>
-      </Box>
-
-      <BottomView
-        graphProperties={propsToInject}
-        datastore={datastore}
-        readonly={debugReadonly}
-        setReadonly={setDebugReadonly}
-      />
-    </Layout>
+    <MockBlockDockProvider
```

Comment:
Wrapped both modes in a provider, so things like logs doesn't get clear when we switch on/off debug mode